### ISSUE TITLE
Errors using azure functionapp on v2.18.0

### ISSUE THREAD

**Issue by tereio** [Azure/azure-cli#16819]
## Describe the bug

**Command Name**
`az functionapp deployment source config-zip`

**Errors:**
```
function() argument 'code' must be code, not str
Traceback (most recent call last):
python3.8/site-packages/knack/cli.py, ln 233, in invoke
    cmd_result = self.invocation.execute(args)
cli/core/commands/__init__.py, ln 558, in execute
    self.commands_loader.load_arguments(command)
azure/cli/core/__init__.py, ln 480, in load_arguments
    self.command_table[command].load_arguments()  # this loads the arguments via reflection
...
python3.8/site-packages/aiohttp/http_parser.py, ln 15, in <module>
    from .helpers import NO_EXTENSIONS, BaseTimerContext
python3.8/site-packages/aiohttp/helpers.py, ln 614, in <module>
    class CeilTimeout(async_timeout.timeout):
TypeError: function() argument 'code' must be code, not str
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az functionapp deployment source config-zip -n {} -g {} --subscription {} -s {} --build-remote {} --src {}`

## Environment Summary
```
macOS-10.15.7-x86_64-i386-64bit
Python 3.8.1
Installer: PIP

azure-cli 2.18.0 *

Extensions:
image-copy-extension 0.2.1

```
## Additional Context

I ran a few tests and found I got this basic error when just calling it directly:
 az functionapp --version                                                                                                                                                         
CLIInternalError: The command failed with an unexpected error. Here is the traceback:
function() argument 'code' must be code, not str
Traceback (most recent call last):
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/knack/cli.py", line 233, in invoke
    cmd_result = self.invocation.execute(args)
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/azure/cli/core/commands/__init__.py", line 584, in execute
    parsed_args = self.parser.parse_args(args)
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/knack/parser.py", line 262, in parse_args
    return super(CLICommandParser, self).parse_args(args)
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/argparse.py", line 1768, in parse_args
    args, argv = self.parse_known_args(args, namespace)
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/azure/cli/core/parser.py", line 289, in parse_known_args
    self._namespace, self._raw_arguments = super().parse_known_args(args=args, namespace=namespace)
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/argparse.py", line 1800, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/argparse.py", line 1988, in _parse_known_args
    positionals_end_index = consume_positionals(start_index)
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/argparse.py", line 1965, in consume_positionals
    take_action(action, args)
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/argparse.py", line 1874, in take_action
    action(self, namespace, argument_values, option_string)
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/argparse.py", line 1159, in __call__
    subnamespace, arg_strings = parser.parse_known_args(arg_strings, None)
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/azure/cli/core/parser.py", line 289, in parse_known_args
    self._namespace, self._raw_arguments = super().parse_known_args(args=args, namespace=namespace)
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/argparse.py", line 1800, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/argparse.py", line 2034, in _parse_known_args
    self.error(_('the following arguments are required: %s') %
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/azure/cli/core/parser.py", line 164, in error
    recommender.set_help_examples(self.get_examples(self.prog))
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/azure/cli/core/parser.py", line 206, in get_examples
    return self.cli_help.get_examples(command,
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/azure/cli/core/_help.py", line 196, in get_examples
    else self.group_help_cls(self, delimiters, parser)
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/knack/help.py", line 254, in __init__
    child.load(options)
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/azure/cli/core/_help.py", line 299, in load
    loader.versioned_load(self, options)
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/azure/cli/core/_help_loaders.py", line 154, in versioned_load
    super(CliHelpFile, help_obj).load(parser)  # pylint:disable=bad-super-call
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/knack/help.py", line 199, in load
    description = getattr(options, 'description', None)
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/knack/parser.py", line 242, in __getattribute__
    self.description = self._description() \
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/azure/cli/core/__init__.py", line 795, in default_description_loader
    op = handler or self.get_op_handler(operation, operation_group=kwargs.get('operation_group'))
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/azure/cli/core/__init__.py", line 829, in get_op_handler
    op = import_module(mod_to_import)
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/azure/cli/command_modules/appservice/custom.py", line 36, in <module>
    from azure.mgmt.storage import StorageManagementClient
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/azure/mgmt/storage/__init__.py", line 9, in <module>
    from ._storage_management_client import StorageManagementClient
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/azure/mgmt/storage/_storage_management_client.py", line 12, in <module>
    from azure.mgmt.core import ARMPipelineClient
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/azure/mgmt/core/__init__.py", line 29, in <module>
    from ._pipeline_client import ARMPipelineClient
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/azure/mgmt/core/_pipeline_client.py", line 26, in <module>
    from azure.core import PipelineClient
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/azure/core/__init__.py", line 30, in <module>
    from ._pipeline_client import PipelineClient
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/azure/core/_pipeline_client.py", line 30, in <module>
    from .pipeline.transport._base import PipelineClientBase
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/azure/core/pipeline/transport/__init__.py", line 60, in <module>
    from ._aiohttp import AioHttpTransport, AioHttpTransportResponse
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/azure/core/pipeline/transport/_aiohttp.py", line 31, in <module>
    import aiohttp
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/aiohttp/__init__.py", line 6, in <module>
    from .client import BaseConnector as BaseConnector
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/aiohttp/client.py", line 35, in <module>
    from . import hdrs, http, payload
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/aiohttp/http.py", line 7, in <module>
    from .http_parser import HeadersParser as HeadersParser
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/aiohttp/http_parser.py", line 15, in <module>
    from .helpers import NO_EXTENSIONS, BaseTimerContext
  File "/Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages/aiohttp/helpers.py", line 614, in <module>
    class CeilTimeout(async_timeout.timeout):
TypeError: function() argument 'code' must be code, not str

**Comment by tereio** [comment]
I've tested down to az cli 2.17.0 and still seeing the same error.  I'll roll back aiohttp and see if this fixes the error.

**Comment by tereio** [comment]
Running:
▶ pip show aiohttp
Name: aiohttp
Version: 4.0.0a1
Summary: Async http client/server framework (asyncio)
Home-page: https://github.com/aio-libs/aiohttp
Author: Nikolay Kim
Author-email: fafhrd91@gmail.com
License: Apache 2
Location: /Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages
Requires: attrs, chardet, multidict, async-timeout, yarl, typing-extensions

**Comment by tereio** [comment]
having rolled this back I now see an error with the knack library:
▶ pip show knack
Name: knack
Version: 0.7.2
Summary: A Command-Line Interface framework
Home-page: https://github.com/microsoft/knack
Author: Microsoft Corporation
Author-email: azpycli@microsoft.com
License: MIT
Location: /Users/bbrew2/.pyenv/versions/3.8.1/lib/python3.8/site-packages
Requires: pygments, tabulate, pyyaml, colorama, jmespath, six, argcomplete
Required-by: azure-cli-core

**Comment by yonzhan** [comment]
functionapp

**Comment by apawast** [comment]
@Hazhzeng @gzuber could either of you please investigate?

**Comment by Flangvik** [comment]
Suddenly also having issues deploying (worked just 40-50 minutes ago)

**Comment by tereio** [comment]
i've done quite a bit of testing on this and it looks like it was due to an upstream dependency that broke the zip deploy remote build step.  Unfortunately the logs I got back gave zero indication of this that I could see.  I only found out by rolling back every change we made on our end until it worked.

The library that did not install correctly in our case was ldap-python.  Presumably due to it needing an internally compiled c lib.

Thanks for looking into this however!
